### PR TITLE
Turn this project into an installable package

### PR DESCRIPTION
**Pull request by Ahmed-Z-D**
This pull request aims to turn this project into an installable package. Turning this project into an installable package can improve its usability and make it easier to maintain. By creating a package, users can easily install the project and its dependencies using the pip package manager, which eliminates the need to manually set up the environment and install each dependency separately.

In addition, creating a package can improve the readability of certain import statements in this project by ensuring that all modules are organized in a logical way within the package. This can make it easier for developers to locate and import the necessary modules, which can save time and effort. For instance, we can change this import statement:
``` 
sys.path.append(os.path.join("..", "googleCloud"))
from initPostgreSQL import initPostgreSQL
```
with this:
``` 
from bettingAI.googleCloud.initPostgreSQL import initPostgreSQL
``` 

Lastly, by turning this project into a package, it becomes easier to add unit tests to all modules within the project. I am planning to use Pytest to write and run tests for each module, ensuring that the code is working as intended and catching any bugs or errors before they become larger issues.

Nevertheless, the content of the requirement.txt file needs to be double-checked, you can review the file and ensure that it includes all required dependencies and their correct versions. You can compare the contents of the requirement.txt file with the dependencies listed in the project’s source code to ensure that they match.

If you notice any discrepancies or errors in the requirement.txt file, you can suggest changes or additions to the file in a comment on this pull request. Be sure to provide clear and detailed explanations for any changes you suggest.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,16 @@
+sqlalchemy  >= 2.0.10
+colorama >= 0.4.6
+pandas >= 2.0.1
+
+# Please confirme that this is the correct library used in the project:
+cloud-sql-python-connector >= 1.2.2
+
+# There are two drivers that are supported for the Postgres dialect, which one of these two is used in the project?
+# cloud-sql-python-connector[pg8000]
```
Comment (erikbohne) [inline requirements.txt:9]: pg8000

**Comment by ASTRAL2** [comment]
I completely agree with you on the importance of having a clear goal and plan for a project, as it helps to keep it on track and ensure that it's progressing toward its intended purpose. In regard to this project, I have no intention of altering its core functionalities. Instead, my plan is to restructure the project in a way that improves its usability, readability, and productivity, while also eliminating any technical debt or code smells. This will involve turning the project into a package, which will make it more organized and accessible for others to use and contribute to. Additionally, I plan to adopt best practices such as automated testing and CI/CD pipelines to ensure the code base remains stable and functional as it evolves over time.

I would be pleased to collaborate with you and walk you through the process of restructuring this project and implementing unit testing.

**Comment by erikbohne** [comment]
Not gonna lie, I like the idea of implementing unit testing. Before merging the branch you've made with the restructure, I have to make sure I solve the merge conflicts as I've made a few changes since the last commit. I will have to allocate some time to take a closer look at the pull request before I decide to merge.

**Comment by Ahmed-Z-D** [comment]
That sounds great! Here is a concise checklist that can help you review this pull request:

- [ ] Ensure that all dependencies are listed in the `requirements.txt` file.
- [ ] Check that the project's dependency versions are up to date and/or compatible with the versions used in your development environment, and update them as necessary to address any security vulnerabilities or bugs.
- [ ] Verify that the `setup.cfg` file contains the same dependencies as listed in `requirements.txt`.
- [ ] Verify that any modified import statements do not cause any code to break by testing the project locally.

As a final tip, it is important to test the changes locally by running the project and verifying that they are functioning as intended. You can install the package in editable mode using the command `pip install -e . `  to ensure that any changes made to the code are reflected in the installed package. Additionally, consider using a virtual environment to isolate project dependencies and ensure consistency across development and deployment environments.

**Comment by erikbohne** [comment]
I've looked through the restructuring and I think that it could benefit the project if you are implementing testing with pytest as you've mentioned. But as the main branch has been updated with quite a lot of code, this pull request has several conflicts. Do you want to change this and update/create a new pull request or should i just restructure it myself?

**Comment by Ahmed-Z-D** [comment]
Assuming there won't be any significant alterations to the code base over the next 48 hours, I'll create a new pull request to prevent any merge conflicts.

**Comment by erikbohne** [comment]
The main branch in my repo is 100% up to date. I will not contribute any more today, so if you make a pull request later today that will be perfect! Closing this pull request and waiting for the new one :)

**Pull request closed without merging**